synfig-core/src/modules/mod_noise/valuenode_random.cpp

```diff
@@ -66,10 +66,14 @@ ValueNode_Random::ValueNode_Random(const ValueBase &value):
 	set_children_vocab(ret);
 	random.set_seed(time(NULL));
 
+	ValueNode_Const::Handle val_smooth;
+	val_smooth = ValueNode_Const::Handle::cast_static(ValueNode_Const::create(int(RandomNoise::SMOOTH_CUBIC)));
+	val_smooth->set_static(true);
+
 	set_link("radius",ValueNode_Const::create(Real(1)));
 	set_link("seed",ValueNode_Const::create(random.get_seed()));
 	set_link("speed",ValueNode_Const::create(Real(1)));
-	set_link("smooth",ValueNode_Const::create(int(RandomNoise::SMOOTH_CUBIC)));
```

Comment:
@rodolforg @FirasH I think better to add optional argument to `ValueNode_Const::create`. Something like:
```
ValueNode_Const() = default;
ValueNode_Const(const ValueBase &x, etl::loose_handle<Canvas> canvas = 0, bool is_static = false);

static ValueNode* create(const ValueBase &x=ValueBase(), etl::loose_handle<Canvas> canvas = 0, bool is_static = false);
```

usage:
```
set_link("smooth",ValueNode_Const::create(int(RandomNoise::SMOOTH_CUBIC), nullptr, true));
```

Reply:
@ice0 how would you set `ValueBase` to static if it's `const`?
Not sure if I'm missing something...

Reply:
https://github.com/synfig/synfig/compare/master...ice0:issue-2131?expand=1

Reply:
>  I think better to add optional argument to `ValueNode_Const::create`.

Hm… I don't know. On other hand the `nullptr, true` as arguments are less 'readable'.

Reply:
> > I think better to add optional argument to `ValueNode_Const::create`.
> 
> Hm… I don't know. On other hand the `nullptr, true` as arguments are less 'readable'.

You can always write something like:
```
ValueNode_Const::create((int)(NORMAL), nullptr, true)); // value, canvas, is_static
```

or 
```
constexpr auto canvas = nullptr;
constexpr auto is_static = true;
ValueNode_Const::create((int)(NORMAL), canvas, is_static));
```

or 
```
ValueNode_Const::create_static((int)(NORMAL));
```

As for me, the real problem is that the method does not return what it should return. It can return `ValueNode_BoneWeightPair` or `ValueNode_Bone` or `ValueNode_Composite` or `ValueNode_Const`.

Reply:
Not sure what's the best way to deal with this...
@ice0 and @rodolforg you're much more experienced than me, I'll let you pick the best solution.

Reply:
I think we can merge this change as is, and back to improvement later. @rodolforg your opinion?

Reply:
> constexpr auto canvas = nullptr;
> constexpr auto is_static = true;
> ValueNode_Const::create((int)(NORMAL), canvas, is_static));

I prefer this versionm if you're using any changes.



> I think we can merge this change as is, and back to improvement later

Ok